src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -83,18 +107,29 @@ public CommandResult execute(Model model) throws CommandException {
 
     /**
      * Creates and returns a {@code Student} with the details of {@code studenttoEdit}
-     * edited with {@code editStudentDescriptor}.
+     * edited with {@code editStudentDescriptor} and {@code editAdminDescriptor}.
      */
-    private static Student createEditedPerson(Student studentToEdit, EditStudentDescriptor editStudentDescriptor) {
+    private static Student createEditedStudent(Student studentToEdit, EditStudentDescriptor editStudentDescriptor,
+                                               EditAdminDescriptor editAdminDescriptor) {
         assert studentToEdit != null;
 
         Name updatedName = editStudentDescriptor.getName().orElse(studentToEdit.getName());
         Phone updatedPhone = editStudentDescriptor.getPhone().orElse(studentToEdit.getPhone());
         School updatedSchool = editStudentDescriptor.getSchool().orElse(studentToEdit.getSchool());
         Year updatedYear = editStudentDescriptor.getYear().orElse(studentToEdit.getYear());
 
-        // Whoever is reworking EditCommand please take note
-        Admin updatedAdmin = editStudentDescriptor.getAdmin().orElse(studentToEdit.getAdmin());
+        ClassTime updatedClassTime = editAdminDescriptor.getClassTime()
+                .orElse(studentToEdit.getAdmin().getClassTime());
+        ClassVenue updatedClassVenue = editAdminDescriptor.getClassVenue()
+                .orElse(studentToEdit.getAdmin().getClassVenue());
+        Fee updatedFee = editAdminDescriptor.getFee()
+                .orElse(studentToEdit.getAdmin().getFee());
+        PaymentDate updatedPaymentDate = editAdminDescriptor.getPaymentDate()
+                .orElse(studentToEdit.getAdmin().getPaymentDate());
+        Set<AdditionalDetail> updatedAdditionalDetails = editAdminDescriptor.getAdditionalDetails()
+                .orElse(studentToEdit.getAdmin().getDetails());
+        Admin updatedAdmin = new Admin(updatedClassVenue, updatedClassTime, updatedFee, updatedPaymentDate,
+                updatedAdditionalDetails);
```

Comment:
Looks good! Perhaps one suggestion would be to call `editAdminDescriptor::isAnyFieldEdited` so we can just create the new student using `student.toEdit().getAdmin()`.

Reply:
Good suggestion! I've implemented it in my latest commit.